src/libstd/thread/mod.rs

```diff
@@ -1052,6 +1054,28 @@ impl<T> JoinInner<T> {
 /// }).unwrap();
 /// ```
 ///
+/// Child being detached and outliving its parent:
+///
+/// ```
+/// use std::thread;
+/// use std::time::Duration;
+///
+/// let original_thread = thread::spawn(|| {
+///     let _detached_thread = thread::spawn(|| {
+///         // Here we sleep to make sure that the first thread returns before.
+///         thread::sleep(Duration::from_millis(10));
+///         // This will be called, even though the JoinHandle is dropped.
+///         println!("♫ Still alive ♫");
+///     });
+/// });
+///
+/// let _ = original_thread.join();
+/// println!("Original thread is joined.");
+/// // We make sure that the child thread has time to run, before the main
+/// // thread returns.
+/// thread::sleep(Duration::from_millis(1000));
```

Comment:
The example could be simplified with a plain scope or function instead of spawning two threads.

There’s no parent-child relationship between the threads in a process.